ValueError: numpy.ndarray size changed

I have the following error when importing `optimum_reparamN2` in my automatic tests (https://github.com/GAA-UAM/scikit-fda/runs/4740232727):
```
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

I suspect that the error is related to the recent release of NumPy 1.22, but I don't know how to solve it. The full trace is shown below:

```
 skfda/__init__.py:37: in <module>
    from . import representation, datasets, preprocessing, exploratory, misc, ml, \
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
skfda/preprocessing/__init__.py:1: in <module>
    from . import registration
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
skfda/preprocessing/registration/__init__.py:9: in <module>
    from ._fisher_rao import ElasticRegistration, FisherRaoElasticRegistration
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
skfda/preprocessing/registration/_fisher_rao.py:16: in <module>
    from ...exploratory.stats import fisher_rao_karcher_mean
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
skfda/exploratory/__init__.py:2: in <module>
    from . import outliers
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
skfda/exploratory/outliers/__init__.py:6: in <module>
    from ._outliergram import OutliergramOutlierDetector
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
skfda/exploratory/outliers/_outliergram.py:8: in <module>
    from ..stats import modified_epigraph_index
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
skfda/exploratory/stats/__init__.py:1: in <module>
    from ._fisher_rao import _fisher_rao_warping_mean, fisher_rao_karcher_mean
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
skfda/exploratory/stats/_fisher_rao.py:7: in <module>
    from fdasrsf.utility_functions import optimum_reparam
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/fdasrsf/__init__.py:24: in <module>
    from .time_warping import fdawarp, align_fPCA, align_fPLS, pairwise_align_bayes, pairwise_align_functions
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/fdasrsf/time_warping.py:9: in <module>
    import fdasrsf.utility_functions as uf
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/fdasrsf/utility_functions.py:21: in <module>
    import optimum_reparamN2 as orN2
src/optimum_reparamN2.pyx:1: in init optimum_reparamN2
    ???
E   ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,5 +42,5 @@ repository = "https://github.com/jdtuck/fdasrsf_python"
 documentation = "https://fdasrsf-python.readthedocs.io/en/latest/"
 
 [build-system]
-requires = ["setuptools>=46.0", "wheel", "cffi>=1.0.0", "Cython", "numpy"]  # PEP 518 - what is required to build
+requires = ["setuptools>=46.0", "wheel", "cffi>=1.0.0", "Cython", "oldest-supported-numpy"]  # PEP 518 - what is required to build
 build-backend = "setuptools.build_meta"

